toonz/sources/toonz/main.cpp

```diff
@@ -666,13 +680,46 @@ int main(int argc, char *argv[]) {
   w.startupFloatingPanels();
 
   CommandManager::instance()->execute(T_Hand);
-  if (!loadScenePath.isEmpty()) {
+  if (!loadFilePath.isEmpty()) {
     splash.showMessage(
-        QString("Loading file '") + loadScenePath.getQString() + "'...",
+        QString("Loading file '") + loadFilePath.getQString() + "'...",
         Qt::AlignCenter, Qt::white);
 
-    loadScenePath = loadScenePath.withType("tnz");
-    if (TFileStatus(loadScenePath).doesExist()) IoCmd::loadScene(loadScenePath);
+    if (loadFilePath.getType() == "toonzscript") {
+      // load script
+      if (TFileStatus(loadFilePath).doesExist()) {
+        // find project for this script file
+        TProjectManager *pm = TProjectManager::instance();
+        TProjectP sceneProject = pm->loadSceneProject(loadFilePath);
+        if (!sceneProject) {
+          std::cerr << QObject::tr(
+                    "It is not possible to load the scene %1 because it does not "
+                    "belong to any project.")
+                    .arg(loadFilePath.getQString()).toStdString() << std::endl;
+          return 1;
+        }
+        if (sceneProject && !sceneProject->isCurrent())
+          pm->setCurrentProjectPath(sceneProject->getProjectPath());
```

Comment:
If OT automatically changes the current project, I think it would be user-friendly to revert it at the end of the session.